Refactor relayable signatures

SuperTux88 opened the pull request:
Step 1 for #64

cmrd-senya commented:
We probably should mark `parent_author_signature` as deprecated in our API reference:
```diff
diff --git a/lib/diaspora_federation/entities/relayable.rb b/lib/diaspora_federation/entities/relayable.rb
index 7a06ddf..21d765b 100644
--- a/lib/diaspora_federation/entities/relayable.rb
+++ b/lib/diaspora_federation/entities/relayable.rb
@@ -45,6 +45,7 @@ module DiasporaFederation
       #   This signature is required only when federating from upstream (parent) post author to
       #   downstream subscribers. This is the case when the parent author has to resend a relayable
       #   received from one of their subscribers to all others.
+      #   @deprecated
       #   @return [String] parent author signature
       #
       # @!attribute [r] parent
```

SuperTux88 commented:
Thanks for the review.

About the `parent_author_signature` in JSON: I removed it completely, so it isn't even added to the JSON when generating it. And since it was never there in JSON (because JSON is new), we don't need it in the schema either. And it doesn't make sense to add it for exporting anyway, because when the exporting user is the parent, we don't need it, and when the exporting user isn't the parent, the `parent_author_signature` can't be included anyway.

I also fixed the other comments, so please have another look at it :)

SuperTux88 commented on docs/_entities/comment.md:
```diff
@@ -50,7 +48,6 @@ See also: [Relayable][relayable]
   <parent_guid>c3893bf029e7013487753131731751e9</parent_guid>
   <text>this is a very informative comment</text>
   <author_signature>cGIsxB5hU/94+rmgIg/Z+OUvXVYcY/kMOvc267ybpk1pT44P1JiWfnI26F1Mta62UjzIW/SjeAO0RIsJRguaISLpXX/d5DJCMpePAZaZiagUbdgH/w4L++fXiPxBKkSm+PB4txxmHGN8FHjwEUJFHJ1m3VfU4w2JC8+IBU93eag=</author_signature>
-  <parent_author_signature>uzjxUSqR8DQBSBa6abY7R/s9DVzT6UAgTctRcUu5rV5o0iXJD2MR6kp6bsVH3nMbbNvOjwAtrdfz3SVHT2gD8M5PmoFagxK7m5T1c9FB0i+wknoAah0Si0c2sP/BPLnkQ83DgLjF+JZCzMX4sWKvYfyaMjnih1MtQILGyuiwA0E=</parent_author_signature>
```
You are right, thanks for the hint. I removed the separation between "From author" and "From parent author" and squashed it to the commit where I already removed `parent_author_signature` from the documentation.
